satpy/composites/__init__.py

```diff
@@ -929,9 +931,10 @@ def build_colormap(palette, dtype, info):
                     for (val, tup) in iterator]
             colormap = Colormap(*tups)
 
-            sf = info.get('scale_factor', np.array(1))
-            colormap.set_range(
-                *info['valid_range'] * sf + info.get('add_offset', 0))
+            if set_range:
+                sf = info.get('scale_factor', np.array(1))
+                colormap.set_range(
+                    *info['valid_range'] * sf + info.get('add_offset', 0))
```

Comment:
How does this work? The `*` at the beginning is expanding a list/tuple but how is `* sf` working properly?

Reply:
```python
In [1]: def foo(*args): 
   ...:     print(args) 
   ...:                                                                                                   

In [2]: import numpy as np                                                                                

In [3]: arr = np.arange(4)                                                                                

In [4]: foo(*arr * 3 - 5)                                                                                 
(-5, -2, 1, 4)
```

Ok, I'll put parenthesis around it for clarity

Reply:
Why is `valid_range` a numpy array? I'm not super comfortable with throwing numpy arrays in Attributes when they aren't needed.

Reply:
well a numpy array here is clearer than a list, since valid range is actually data values that we can have to scale or manipulate as data. But I can cast valid_range to a numpy array if that makes you feel better :)

Reply:
If the attribute is `valid_range` then it should be two elements following the CF standard (or convention) shouldn't it? A tuple or list (one works better with xarray iirc) makes the most sense in that case.

Reply:
I don't really see what makes a tuple or a list better than a numpy array, but I'll remove  the assumption that it is.

Reply:
I think of a DataArray as being an in-memory representation of a NetCDF file Variable. Storing attributes that aren't builtin types seems odd to me. Especially if there are two-elements that are typically represented as a list/tuple when read from a NetCDF file.

Reply:
xarray reads valid_range as an array:
```python
In [1]: import xarray as xr                                                                                                      

In [2]: filename = '/data/temp/AdamD/CMProb/S_NWC_CPP_noaa20_11508_20200207T1120275Z_20200207T1121520Z.nc'                       

In [3]: nc = xr.open_dataset(filename, decode_cf=True, mask_and_scale=False, chunks=4096)                                        

In [4]: nc['cpp_cot'].attrs['valid_range']                                                                                       
Out[4]: array([    0, 32000], dtype=uint16)

In [5]: type(nc['cpp_cot'].attrs['valid_range'])                                                                                 
Out[5]: numpy.ndarray
```